src/main/java/com/google/devtools/build/lib/skyframe/TreeArtifactValue.java

```diff
@@ -60,7 +61,7 @@ public PathFragment apply(Artifact artifact) {
    * Returns a TreeArtifactValue out of the given Artifact-relative path fragments
    * and their corresponding FileArtifactValues.
    */
-  static TreeArtifactValue create(Map<TreeFileArtifact, FileArtifactValue> childFileValues) {
+  static TreeArtifactValue create(SortedMap<TreeFileArtifact, FileArtifactValue> childFileValues) {
     Map<String, FileArtifactValue> digestBuilder =
```

Comment:
I think this map also needs to be sorted in order to keep the digest stable.

Reply:
Ah, we xor inside the metadata calculation so this doesn't need to be sorted.

Reply:
Alright - FYI: I will have a look again into the test for it tomorrow I guess

Reply:
There's actually an internal change pending that has the same idea as this change, but the author will be out until Monday. You can hold off on this change, or put a test in TreeArtifactMetadataTest (which is where the internal change has a test). Should be simple.

Reply:
@janakdr I added a test as suggested to `TreeArtifactMetaDataTest` and did not change the Map for the digestBuilder.